### DISCUSSION TITLE
Swiper in node module complains that: An import path cannot end with a '.d.ts' extension

### DISCUSSION THREAD

**Discussion by JustinPraas** [nolimits4web/swiper#6829] [Q&A]
I have installed Swiper in my NextJS project, but for some reason I can't get it to work properly.

`tsconfig`
```{
  "compilerOptions": {
    "target": "es5",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true,
    "plugins": [
      {
        "name": "next"
      }
    ]
  },
  "include": [
    "next-env.d.ts",
    "**/*.ts",
    "**/*.tsx",
    ".next/types/**/*.ts",
    "lib/mongodb.ts",
  ],
  "exclude": [
    "node_modules"
  ]
}
``` 

Versions:
Next: 13.4.8
React: 18.2.0
Swiper: 10.0.3

The error that I am getting:
![image](https://github.com/nolimits4web/swiper/assets/26166921/f1659970-0c75-4e7b-ad38-6e2e4b07403c)
![image](https://github.com/nolimits4web/swiper/assets/26166921/aa81a83a-cfed-4bae-abe3-0ebaea1c1f92)

Has anyone experienced something like this before? Am I missing some TypeScript configuration?

**Comment by nolimits4web** [comment]
Try setting moduleResolution to nodenext

**Reply by JustinPraas** [reply]
That produces different compilation errors regarding other imports unrelated to Swiper

**Comment by JustinPraas** [comment]
I don't know what I did, but the import errors in node_modules have gone away. Probably an IDE issue... However, I'm still stuck with the following error:

```
Type error: 
Type '{ children: Element[] | undefined; spaceBetween: number; slidesPerView: string; pagination: boolean; speed: number; autoplay: { delay: number; disableOnInteraction: boolean; }; draggable: false; modules: SwiperModule[]; }' is not assignable to type 'IntrinsicAttributes & RefAttributes<SwiperRef> & SwiperProps'.

Property 'spaceBetween' does not exist on type 'IntrinsicAttributes & RefAttributes<SwiperRef> & SwiperProps'
```

**Comment by JustinPraas** [comment]
The components seem to work in development, where I don't get the error. It only happens when I build my next application during type checking

**Comment by ReallyBean** [comment]
same error using `yarn build`
![Screenshot 2023-11-09 at 22 28 01](https://github.com/nolimits4web/swiper/assets/40273513/79300091-b3b9-40d9-b401-f4da03dead50)

**Reply by Anubarak** [reply]
were you able to solve it?

**Comment by EmilZapata** [comment]
were you able to solve it?

**Comment by EliezerB123** [comment]
We fixed this exact error in our project (Angular 16) by:
1. Deleting the node_modules folder.
2. Upgrading typescript in our `package.json` from `4.9.5` to `5.1.6`.

**Reply by xiaolidu111** [reply]
thank you, i slove this problem,you are my father!
